packages/react-server/src/ReactFlightServer.js

```diff
@@ -4058,12 +4091,25 @@ function renderDebugModel(
 
     if (counter.objectLimit <= 0 && !doNotLimit.has(value)) {
       // We've reached our max number of objects to serialize across the wire so we serialize this
-      // as a marker so that the client can error when this is accessed by the console.
-      return serializeLimitedObject();
+      // as a marker so that the client can error or lazy load thiswhen accessed by the console.
+      return serializeDeferredObject(request, value);
     }
 
     counter.objectLimit--;
 
+    const deferredDebugObjects = request.deferredDebugObjects;
+    if (deferredDebugObjects !== null) {
+      const deferredId = deferredDebugObjects.existing.get(value);
+      // We earlier deferred this same object. We're now going to eagerly emit it so let's emit it
+      // at the same ID that we already used to refer to it.
```

Comment:
Why do we emit eagerly when we see it twice?

Reply:
We only get here if we have passed the objectLimit check above. Either because something else with a high limit also wrote this object or because we're now asking for it from the client.

If there's still an object limit being enforced we wouldn't get here no matter how many times we see the same object.